vulkaninfo/vulkaninfo.cpp

```diff
@@ -853,10 +864,29 @@ int main(int argc, char **argv) {
     }
 #endif
 
+    int return_code = 0;  // set in case of error
     uint32_t selected_gpu = 0;
     bool show_formats = false;
     char *output_path = nullptr;
 
+    // Figure out the name of the executable, pull out the name if given a path
+    // Default is `vulkaninfo`
+    std::string executable_name = "vulkaninfo";
+    if (argc >= 1) {
+#ifdef _WIN32
+        const char *platform_separator = "\\";
+#else
+        const char *platform_separator = "/";
+#endif
+        executable_name = std::string(argv[0]);
+        // don't include path separator
+        auto last_occurrence = executable_name.rfind(platform_separator);
+        if (last_occurrence != std::string::npos && last_occurrence < executable_name.length()) {
```

Comment:
What does `last_occurrence < executable_name.length()` gain us, if we know we're not searching for an empty string?

Reply:
Redid the code, should be better and more legible. My previous point still stands however. Need to check both if its npos and if its less than the size/length.